packages/scripts/scripts/test-e2e.js

```diff
@@ -27,6 +28,14 @@ const {
 	hasJestConfig,
 } = require( '../utils' );
 
+const result = spawn( 'node', [ require.resolve( 'puppeteer/install' ) ], {
+	stdio: 'inherit',
+} );
```

Comment:
Not something where I think any change will be warranted, but:

I would always prefer to use a module's API than to spawn a process, when one is offered. Puppeteer does have its [`browserFetcher`](https://github.com/puppeteer/puppeteer/blob/v2.1.1/docs/api.md#puppeteercreatebrowserfetcheroptions). I guess the main issue is that [Puppeteer's install script does a lot of up-front work](https://github.com/puppeteer/puppeteer/blob/0b1777e73cb1e83ece9e09b7b51d11b798def06f/install.js#L38-L52) to decide if and which Chromium to download.

So, from our perspective, this is still probably the easiest (most direct) way to go about doing it.

Reply:
That said, I wonder if, once you take away all of the options that Puppeteer is accounting for, if it's really that much trouble:

```js
const puppeteer = require( 'puppeteer' );
const fetcher = puppeteer.createBrowserFetcher();
const revision = require( 'puppeteer/package.json' ).puppeteer.chromium_revision;
if ( ! fetcher.revisionInfo( revision ).local ) {
    await fetcher.download( revision );
}
```

([In the future](https://v8.dev/features/top-level-await), at least)

Reply:
We can seek for optimizations once I sort out how to make all tests pass :)

I plan to open another PR where I upgrade `puppeteer` to the latest version to make sure it has an identical API as `puppeteer-core`.